### PR TITLE
Add support for ESLint v4 to package.json

### LINKED ISSUE #196: Update ESLint from 3.x to 4.x
Currently gulp-eslint package.json only installs eslint v3. With the soon to be released eslint v4 it would be nice if gulp-eslint could be upgraded as well. Do we need to have eslint as dependency? Would a peer dependency work as well?

### PR DESCRIPTION
**Pull request by soda0289**
Fix #196

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
   "license": "MIT",
   "dependencies": {
     "bufferstreams": "^1.1.1",
-    "eslint": "^3.0.0",
+    "eslint": "3.x.x - 4.x.x",
```
Comment (shinnn) [inline package.json]: `^4.0.0`

Comment (soda0289) [inline package.json]: They have only released alpha build which means settings it to ^4.0.0 will not work. I can wait a few weeks until the official v4 is out. Thanks.

**Comment by soda0289** [comment]
Will wait for official v4 to be released

**Comment by shinnn** [comment]
@soda0289 No need to close this. We can keep this open and will merge it when `eslint` 4.0.0 is available.

**Comment by soda0289** [comment]
I screwed up by force pushing to this branch after closing sorry it. Sorry, I'll have to open another PR

**Pull request closed without merging**